### ISSUE TITLE
Raw data not being submitted, and then no data can be viewed on the website

### ISSUE THREAD

**Issue by teravestdan** [venturit/photosynq-app#53]
Sometimes, the JSON data gets cut off after "submitted" without any of the raw data being attached to the datum ID (see below). When this happens no data is visible on the data explorer for the project.

{"datum_id":94359,"datum_url":"http://staging.photosynq.venturit.net/projects/panicum-capillare-survey/explore_single_measurement?datum_id=94359","user_id":482,"status":"submitted"},{"datum_id":94358,"...

**Comment by gbathree** [comment]
yes, this is really important!  This bug is confirmed to come only from the Android app and not chrome app, so I believe it occurs at the Android app level not at the server.

**Comment by galacto** [comment]
We found that it happens when device turns off and turns on again in between measurements. 
i.e. multispeq send {pwr_off: "high"} and gets submitted.

**Comment by kalpeshdeo** [comment]
In Android app we are checking the pwr_off message, if its pwr_off then app will not submit the data.

**Comment by gbathree** [comment]
Checking for pwr_off is a good short term fix - but this is a good long term fix:

posted @108 also:

In order to accommodate a wide variety of input data types, but not accept unintended data (like my pwr_off JSONs coming from the device), we came up with the following rules for data screening:

All data must have time stamp
All data must have -
  - EITHER - "sample" key
  - OR - "user_questions" key

This implies data has either sensor data and no survey data, OR survey data and no sensor data, OR both survey and sensor data.  If it has neither then we're not interested :)  And all data should have a timestamp!

This data screen should occur both at the app levels AND at the database level.  This is to ensure that lots of bad data does not get cached, and then produce a bunch of server errors which is visible to the user, but also so that other (non-MultispeQ) data source coming into the platform are also screened before being saved.  This is an important structural component of the platform as we expand and begin accepting foreign devices onto the platform.
